src/main/java/ankel/dropbear/impl/InvocationHandler.java

```diff
@@ -214,46 +211,61 @@ private HttpUriRequest setEntityForHttpRequest(
 
     baseRequest.setHeader(HttpHeaders.CONTENT_TYPE, contentTypeHeader);
 
-    Object entityPojo = UriBuilderUtils.getEntityObject(method, args);
-
     final HttpEntity httpEntity;
 
-    if (entityPojo instanceof InputStream)
-    {
-      httpEntity = new InputStreamEntity(
-          (InputStream)entityPojo, ContentType.create(contentTypeHeader));
-    }
-    else
+    if (contentTypeHeader.equals(MediaType.APPLICATION_JSON) || contentTypeHeader.equals(MediaType.TEXT_PLAIN))
     {
-      RestClientSerializer serializer = null;
 
-      for (RestClientSerializer s : restClientSerializers)
+      final Object entityPojo = UriBuilderUtils.getEntityObject(method, args);
+
+      if (entityPojo instanceof InputStream)
       {
-        if (s.getSupportedMediaTypes().contains(contentTypeHeader))
+        httpEntity = new InputStreamEntity(
+            (InputStream) entityPojo, ContentType.create(contentTypeHeader));
+      }
+      else
+      {
+        RestClientSerializer serializer = null;
+
+        for (RestClientSerializer s : restClientSerializers)
         {
-          serializer = s;
-          break;
+          if (s.getSupportedMediaTypes().contains(contentTypeHeader))
+          {
+            serializer = s;
+            break;
+          }
         }
-      }
 
-      if (serializer == null)
-      {
-        throw new IllegalStateException(
-            String.format("Cannot find a suitable serializer for method %s#%s",
-                method.getDeclaringClass().getCanonicalName(), method.getName()));
-      }
+        if (serializer == null)
+        {
+          throw new IllegalStateException(
+              String.format("Cannot find a suitable serializer for method %s#%s",
+                  method.getDeclaringClass().getCanonicalName(), method.getName()));
+        }
 
-      try
-      {
-        httpEntity = new StringEntity(
-            serializer.serialize(entityPojo),
-            ContentType.create(contentTypeHeader));
-      }
-      catch (Exception e)
-      {
-        throw new RuntimeException("Failed to serialize entity", e);
+        try
+        {
+          httpEntity = new StringEntity(
+              serializer.serialize(entityPojo),
+              ContentType.create(contentTypeHeader));
+        }
+        catch (Exception e)
+        {
+          throw new RuntimeException("Failed to serialize entity", e);
+        }
       }
     }
+    else if (contentTypeHeader.equals(MediaType.APPLICATION_FORM_URLENCODED))
+    {
+      // Get the Name/Value Pairs
+
+      final List<NameValuePair> nameValuePairs = UriBuilderUtils.getFormParameters(method, args);
+      httpEntity = EntityBuilder.create().setParameters(nameValuePairs).build();
+    }
+    else
+    {
+      throw new RuntimeException(String.format("Content Type [%s] is not supported at this time", contentTypeHeader));
```

Comment:
Different content types are supported (see the test for TEXT/PLAN type) Flip this if around like this  


```java
If (contentTypeHeader.equals(MediaType.APPLICATION_FORM_URLENCODED)) 
// new code here
else
// old code here
```